ui/jobs/utils.ts

```diff
@@ -214,3 +181,17 @@ export const makeAuraTimerIcon = (
 
   return div;
 };
+
+export const normalizeLogLine = <Fields extends NetAnyFields>(
+  line: string[],
+  fields: Fields,
+): Partial<Record<keyof Fields, string>> => {
+  return new Proxy({}, {
+    get(_target, property) {
+      if (typeof property === 'string' && property in fields) {
+        const fieldKey = fields[property as keyof Fields];
+        return line[fieldKey as unknown as number];
```

Comment:
```suggestion
        const looseFields: { [prop: string]: number } = fields;
        const fieldKey: number | undefined = looseFields[property];
        if (fieldKey)
          return line[fieldKey];
```

Suggestion for fewer type assertions.